packages/amplify_datastore/ios/Classes/types/query/QueryPredicateBuilder.swift

```diff
@@ -19,14 +19,25 @@ import Amplify
 
 public class QueryPredicateBuilder {
     
+    // Remove model name if queryPredicate is on id 
+    private static func extractFieldName( field : String? ) -> String? {
+        
+        if(field != nil && field!.hasSuffix(".id")){
+            return "id"
+        }
+        return field;
+    }
+    
```

Comment:
I thought we decided to not add the prefix in dart and add it in android only.

Reply:
Hmm I thought we decided to have the prefix automatically added in the codegen models

![image](https://user-images.githubusercontent.com/11478656/99842767-49fc7c00-2b25-11eb-9e5a-f95587734d04.png)

Reply:
hmm there seems to be some confusion around that. What I recalled, there were two options
1. Add the prefix in dart which will work with android and remove this prefix in iOS
2. Don't add the prefix in dart, only add it in android code, iOS left untouched.

I thought we were going with 2, I'm ok either way